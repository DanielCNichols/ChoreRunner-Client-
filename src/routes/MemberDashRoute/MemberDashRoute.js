import React, { useState, useEffect } from 'react';
import ApiService from '../../services/api-service.js';
import s from './MemberDashRoute.module.css';
import Leaderboard from '../../components/LeaderBoard/LeaderBoard';
import PlayerStats from '../../components/PlayerStats/PlayerStats';

export default function MemberDashRoute() {
  const [assignedTasks, setAssignedTasks] = useState([]);
  const [completedTasks, setCompletedTasks] = useState([]);
  const [rankings, setRankings] = useState([]);
  const [userStats, setUserStats] = useState({});
  const [error, setError] = useState(null);

  useEffect(() => {
    ApiService.getMemberStatus()
      .then(({ assignedTasks, completedTasks, userStats, rankings }) => {
        setAssignedTasks(assignedTasks);
        setCompletedTasks(completedTasks);
        setUserStats(userStats);
        setRankings(rankings);
      })
      .catch(error => setError(error));
  }, []);

  function updateTasks(id) {
    let newTasks = [...assignedTasks];
    let index = newTasks.findIndex(task => task.id === id);
    newTasks[index].status = 'completed';

    setAssignedTasks(newTasks);
  }

  async function handleCompleted(id) {
    try {
      await ApiService.completeTask(id);
      updateTasks(id);
    } catch (error) {
      setError(error);
    }
  }

  function TaskItem({ task: { id, title, points, status } }) {
    return (
      <li className={s.taskItem}>
        <div className={s.taskName}>
          <p>{title}</p>
        </div>
        <div className={s.points}>
          <span>{points} EXP</span>
        </div>
        <div className={s.taskControl}>
          {status === 'assigned' ? (
            <button
              className="arcadeButton"
              onClick={() => {
                handleCompleted(id);
              }}
            >
              Clear!
            </button>
          ) : (
            <p className={s.waiting}>Awaiting Approval!</p>
          )}
        </div>
      </li>
    );
  }

  return (
    <section className={s.memberDashboard}>
      <div className={s.memberDashWrapper}>
        <header>
          <h2 className="videoGameTitles">{userStats.name}'s Profile</h2>
        </header>
        <Leaderboard rankings={rankings} />
        <PlayerStats userStats={userStats} />
        <div className={s.choresContainer}>
          <h2>Chore-llenges</h2>

          {!assignedTasks.length && !completedTasks.length && (
            <p className={s.taskAlert}>You don't have any chores to do!</p>
          )}
          {assignedTasks.length ? (
            <ul>
              {assignedTasks.map(task => {
                return <TaskItem key={task.id} task={task} />;
              })}
            </ul>
          ) : null}

          {completedTasks.length ? (
            <ul>
              {completedTasks.map(task => {
                return <TaskItem key={task.id} task={task} />;
              })}
            </ul>
          ) : null}
        </div>
      </div>
    </section>
  );
}
