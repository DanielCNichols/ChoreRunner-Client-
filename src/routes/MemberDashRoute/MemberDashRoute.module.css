.memberDashboard {
  box-sizing: border-box;
  height: 100%;
  width: 100%;
  position: relative;
  top: 10vh;
  padding: 0;
  background-image: url('../../ImgAssets/ChoreRunner.png');
  background-size: cover;
  background-position: 0 0;
  background-repeat: repeat;
  animation: house 45s linear infinite;
}

.memberDashWrapper {
  width: 100%;
  padding: 25px 10px 100px 10px;
}

.memberDashboard header {
  padding: 20px;
  background: rgba(0, 0, 0, 0.7);
  border-radius: 6px;
  border: 3px solid white;
  margin: auto;
}

.memberDashboard header h2 {
  font-size: 1.2em;
}

.choresContainer {
  background: rgba(0, 0, 0, 0.7);
  border-radius: 10px;
  width: 95%;
  margin: 20px auto;
  padding-bottom: 20px;
  border: 3px solid white;
}

.points span {
  color: white;
  font-size: 18px;
  font-family: 'Electrolize', sans-serif;
}
.taskName p {
  color: white;
  font-size: 30px;
  font-family: 'Electrolize', sans-serif;
}

.taskAlert {
  color: white;
  font-size: 20px;
  margin: 0;
  font-family: 'Electrolize', sans-serif;
}

.choresContainer h2 {
  padding-top: 20px;
  color: rgb(255, 217, 0);
  font-size: 20px;
  text-shadow: none;
  font-family: 'Press Start 2P', cursive;
}

.task-control button {
  text-align: center;
  margin: 20px 5px;
  font-family: 'Press Start 2P', cursive;
  font-size: 13px;
  padding: 10px;
  background-color: rgb(255, 217, 0);
  text-shadow: -0.5px -1px rgb(105, 90, 0), 1px 1px white;
  color: #725f00;
  -webkit-border-radius: 7px;
  -moz-border-radius: 7px;
  -o-border-radius: 7px;
  border-radius: 7px;
  box-shadow: 0 0.3em rgb(134, 112, 0);
  cursor: pointer;
  animation: bounce 500ms infinite alternate;
  -webkit-animation: bounce 500ms infinite alternate;
  cursor: pointer;
}

@keyframes house {
  from {
    background-position: 0 0;
  }
  to {
    background-position: -13690px 0;
  }
}

@media screen and (min-width: 470px) {
  .choresContainer h2 {
    padding-top: 20px;
    color: rgb(255, 217, 0);
    font-size: 20px;
    text-shadow: none;
    font-family: 'Press Start 2P', cursive;
  }

  .member-container {
    position: absolute;
    width: 100%;
  }

  @keyframes house {
    from {
      background-position: 0 0;
    }
    to {
      background-position: -13690px 0;
    }
  }

  .points {
    /* margin: 0 20px; */
    min-width: 80px;
  }

  .taskName {
    min-width: 250px;
  }
}
@media screen and (min-width: 500px) {
  .memberDashboard {
    top: 5vh;
    padding-top: 50px;
    overflow-y: auto;
    background-image: url('../../ImgAssets/ChoreRunner.png');
    background-size: cover;
    background-position: 0 0;
    background-repeat: repeat;
    animation: house 45s linear infinite;
  }

  .memberDashWrapper {
    padding-top: 50px;
    width: 80%;
    margin: auto;
  }

  .memberDashboard header {
    width: 80%;
  }

  .choresContainer {
    background: rgba(0, 0, 0, 0.7);
    border-radius: 10px;
    margin-bottom: 100px;
    width: 80%;
    margin: 20px auto;
  }
}

@media screen and (min-width: 850px) {
  .choresContainer {
    width: 60%;
  }

  .badge-task-container {
    display: flex;
    flex-direction: column;
    width: 80%;
    margin: 0 auto 100px auto;
    justify-content: space-between;
  }
}

@media screen and (min-width: 1050px) {
  .memberDashWrapper {
    width: 60%;
  }
  .choresContainer {
    width: 80%;
    background-color: blue;
  }
}
