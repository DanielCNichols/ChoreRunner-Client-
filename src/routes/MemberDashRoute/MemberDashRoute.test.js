import React from 'react';
import ReactDOM from 'react-dom';
import MemberDashRoute from './MemberDashRoute';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('MemberDashboard component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <MemoryRouter>
        <MemberDashRoute />
      </MemoryRouter>,
      div
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <MemberDashRoute />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
