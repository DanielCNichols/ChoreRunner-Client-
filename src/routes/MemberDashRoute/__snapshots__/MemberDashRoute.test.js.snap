// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberDashboard component testing renders UI as expected 1`] = `
<section
  className="memberDashboard"
>
  <div
    className="memberDashWrapper"
  >
    <header>
      <h2
        className="videoGameTitles"
      >
        's Profile
      </h2>
    </header>
    <section
      className="leaderBoard"
    >
      <h3>
        Leaderboard
      </h3>
      <div
        className="leaderBoardGrid"
      >
        <p>
          Rank
        </p>
        <p>
          Member Name
        </p>
        <p>
          Score
        </p>
      </div>
      <ul
        className="rankings"
      />
    </section>
    <section
      className="levelInfoGrid"
    >
      <h3
        className="videoGameTitles"
      >
        's Stats
      </h3>
      <div
        className="badge"
      >
        <div
          className="imageContainer"
        >
          <img
            alt="Badge.png"
          />
        </div>
        <p>
          Level 
        </p>
      </div>
      <div
        className="status"
      >
        <h3>
          Experience
        </h3>
        <div
          className="progressBarContainer"
        >
          <div
            style={
              Object {
                "backgroundColor": "blue",
                "border": "2px solid white",
                "height": "30px",
                "position": "relative",
                "width": "100%",
              }
            }
          >
            <span
              style={
                Object {
                  "color": "#ffd900",
                  "display": "inline-block",
                  "fontWeight": "bold",
                  "padding": "5px",
                  "position": "relative",
                  "textAlign": "center",
                  "verticalAlign": "middle",
                  "width": "100%",
                  "zIndex": "2",
                }
              }
            >
              NaN / 10
            </span>
            <div
              style={
                Object {
                  "backgroundColor": "white",
                  "height": "100%",
                  "position": "absolute",
                  "textAlign": "right",
                  "top": "0",
                  "width": "NaN%",
                }
              }
            />
          </div>
        </div>
        <p>
          Total EXP: 
          <span />
        </p>
        <p>
          EXP to next level: 
          <span />
        </p>
      </div>
    </section>
    <div
      className="choresContainer"
    >
      <h2>
        Chore-llenges
      </h2>
      <p
        className="taskAlert"
      >
        You don't have any chores to do!
      </p>
    </div>
  </div>
</section>
`;
